tests/test_model.py

```diff
@@ -481,6 +481,7 @@ def assert_dict_lists_equal(self, list1, list2):
                 if d2_item == d1_item:
                     found = True
             if not found:
+                assert list1 == list2
```

Comment:
Can we just drop this function entirely with pytest? Probably worth doing a sweep as a separate PR...